fvm/src/syscalls/error.rs

```diff
@@ -50,13 +51,33 @@ impl Abort {
     }
 }
 
-/// Wraps an execution error in a Trap.
+impl fmt::Display for Abort {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        use Abort::*;
+
+        // todo make this less bad
+        let desc = match self {
+            Exit(_, _, _) => "exit",
+            OutOfGas => "out of gas",
+            Fatal(_) => "fatal error",
+        };
+        write!(f, "wasm trap: {desc}")
+    }
+}
+
+impl std::error::Error for Abort {
```

Comment:
Is there a better way to do that? Or at least a way to get rid of the manual Display impl above?